components/brave_new_tab_ui/containers/newTab/sponsored_rich_media_background.tsx

```diff
@@ -0,0 +1,109 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import styled from 'styled-components'
+
+export interface SponsoredRichMediaBackgroundInfo {
+  url: string
+  creativeInstanceId: string
+  placementId: string
+  targetUrl: string
+}
+
+interface StatusProps {
+  sponsoredRichMediaBackgroundHasLoaded: boolean
+}
+
+interface Props extends StatusProps {
+  sponsoredRichMediaBackgroundInfo: SponsoredRichMediaBackgroundInfo
+  onEventReported: (name: NewTab.SponsoredRichMediaEventType) => void
+  onLoaded: () => void
+}
+
+const iframeAllow = `
+  accelerometer 'none';
+  ambient-light-sensor 'none';
+  camera 'none';
+  display-capture 'none';
+  document-domain 'none';
+  fullscreen 'none';
+  geolocation 'none';
+  gyroscope 'none';
+  magnetometer 'none';
+  microphone 'none';
+  midi 'none';
+  payment 'none';
+  publickey-credentials-get 'none';
+  usb 'none'
+`
+
+const SponsoredRichMediaBackgroundIframe = styled('iframe') <StatusProps>`
+  opacity: ${p => p.sponsoredRichMediaBackgroundHasLoaded ? 1 : 0};
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  border: none;
+  z-index: 0;
+`
+
+/// We expect the event data to be of the following format:
+/// {
+///   type: 'richMediaEvent',
+///   value: 'click'
+/// }
+function getEventType(event: MessageEvent): NewTab.SponsoredRichMediaEventType | undefined {
+  if (!event.data || event.data.type !== 'richMediaEvent') {
+    return undefined
+  }
+
+  if (event.data.value === 'click' || event.data.value === 'mediaPlay'
+      || event.data.value === 'media25' || event.data.value === 'media100') {
+    return event.data.value as NewTab.SponsoredRichMediaEventType
```

Comment:
Since you've narrowed the type of `event.data.value`, can you just return without the `as` type assertion? That way the compiler will complain if you add another value to the `SponsoredRichMediaEventType` enum.